not_tcp/host.py

```diff
@@ -96,3 +99,73 @@ def from_bytes(cls, buf: bytes) -> (Optional["Packet"], bytes):
         body = body_and_remainder[:header.body_length]
         remainder = body_and_remainder[header.body_length:]
         return (Packet.from_header(header, body), remainder)
+
+
+# Use as superclass; subclass to simulator or real
+class StreamProxy:
+    lock = asyncio.Lock()
+
+    def send(self, b: bytes()):
+        # Must be implemented by subclass
+        pass
+
+    def recv(self) -> bytes:
+        # Must be implemented by subclass
+        pass
+
+    def client_connected(
+            self, reader: StreamReader, writer: StreamWriter):
+        asyncio.create_task(self.client_loop(reader, writer))
+
+    async def client_loop(self, reader: StreamReader, writer: StreamWriter):
+        async with self.lock, asyncio.TaskGroup() as tg:
+            tg.create_task(self.run_inbound(reader))
+            tg.create_task(self.run_outbound(writer))
+
+    async def run_inbound(self, reader: StreamReader):
+        p1 = Packet(flags=Flag.START, stream_id=1, body=bytes())
+        self.send(p1.to_bytes())
+        want_bytes = 256
+        while True:
+            try:
+                async with asyncio.timeout(1):
+                    buffer = await reader.read(want_bytes)
+                    if len(buffer) == 0:
+                        # Zero bytes returned at EOF; but not a timeout.
+                        # That's end-of-stream.
+                        break
+                    # On a successful read, keep that many bytes
+                    want_bytes = 256
+                    p2 = Packet(flags=0, stream_id=1, body=buffer)
+                    self.send(p2.to_bytes())
+            except asyncio.TimeoutError:
+                want_bytes = want_bytes // 2
+                if want_bytes == 0:
+                    want_bytes = 1
+        # Input is done, in theory
+        p3 = Packet(flags=Flag.END, stream_id=1, body=bytes())
+        self.send(p3.to_bytes())
```

Comment:
This seems to be the cause of the issue where we can't send more than one request.  If I get rid of this line, I can get as many /coffee (418) and /counts responses I want from curl

Reply:
The rewrite/split of "separate state machines" also fixed the "only one request", without getting rid of this -- and I think the "stop" is important to capture, so I'd like to leave this.

Merging this with the bug intact, #39 has the rewrite.